### PR TITLE
Add docker file for CXX build and build_cxx script to build the CXX sdk.

### PR DESCRIPTION
**Pull request by cianx**
Signed-off-by: Cian Montgomery <cian.montgomery@intel.com>

### CODE REVIEW

File: docker/sawtooth-dev-cxx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+# Description:
+#   Builds an image to be used when developing in Python. The default CMD is to
+#   run build_python.
```
Comment (jjason) [inline docker/sawtooth-dev-cxx]: Python??? or C++????

File: docker/sawtooth-dev-cxx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+#
+#   The image is also used by unit and integration tests. Docker is used to
+#   interact with and orchestrate the creation of other docker containers.
+#
+# Build:
+#   $ cd sawtooth-core/docker
+#   $ docker build . -f sawtooth-dev-python -t sawtooth-dev-python
```
Comment (jjason) [inline docker/sawtooth-dev-cxx]: cxx?

File: docker/sawtooth-dev-cxx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#
+# Run:
+#   $ cd sawtooth-core
+#   $ docker run -v $(pwd):/project/sawtooth-core \
+#     -v /var/run/docker.sock:/var/run/docker.sock sawtooth-dev-python
```
Comment (jjason) [inline docker/sawtooth-dev-cxx]: cxx?

File: docker/sawtooth-dev-cxx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    https://github.com/google/protobuf/releases/download/v3.2.0/protobuf-cpp-3.2.0.zip \
+ && unzip protobuf-cpp-3.2.0.zip \
+ && cd protobuf-3.2.0 \
+ && ./configure \
+ && make \
+ && make check \
+ && make install \
+ && ldconfig \
+ && cd .. \
+ && rm -rf cd protobuf-3.2.0
```
Comment (aludvik) [inline docker/sawtooth-dev-cxx]: Can this be packaged up as a deb and moved to the package repo so it is just an install? This pushes up the build time substantially for a fresh build.

Comment (cianx) [inline docker/sawtooth-dev-cxx]: @aludvik sure, no idea how to do that. I have a story open in jira, several sprints. (https://jira.hyperledger.org/browse/STL-167)

File: docker/sawtooth-dev-cxx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+# Description:
+#   Builds an image to be used when developing in C++. The default CMD is to
+#   run build_cxx.
+#
+#   The image is also used by unit and integration tests. Docker is used to
+#   interact with and orchestrate the creation of other docker containers.
+#
+# Build:
+#   $ cd sawtooth-core/docker
+#   $ docker build . -f sawtooth-dev-python -t sawtooth-dev-cxx
```
Comment (jjason) [inline docker/sawtooth-dev-cxx]: You missed a `python`

File: docker/sawtooth-dev-cxx (hunk 4 above)
Comment (jsmitchell) [inline docker/sawtooth-dev-cxx]: Can we disable the long builds by default until the packaging story is done?

File: docker/sawtooth-dev-cxx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+# Run:
+#   $ cd sawtooth-core
+#   $ docker run -v $(pwd):/project/sawtooth-core sawtooth-dev-cxx
+
+FROM ubuntu:xenial
+
+LABEL "install-type"="mounted"
+
+RUN echo "deb http://repo.sawtooth.me/ubuntu/ci xenial universe" >> /etc/apt/sources.list \
+ && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 6B58B1AC10FB5F63 \
```
Comment (aludvik) [inline docker/sawtooth-dev-cxx]: This key changed to `8AA7AF1F1091A5FD`

**Comment by aludvik** [comment]
This needs to be rebuilt once the protobuf deb is published to ubuntu/ci

**Comment by cianx** [comment]
Verified docker and vagrant build locally.

**Pull request merged**